### ISSUE TITLE
After update to 0.5.81 some windows 7 agents change from online to offline status continuously

### ISSUE THREAD

**Issue by edorbike** [Ylianst/MeshCentral#1955]
Some windows 7 [7601] 32bit and 64 bit devices continously change from online to offline. Windows 10 devices work properly.

Restarting mesh client agent service on a remote pc change status to online but some minutes or hours later change to offline again, however, I can access from teamviewer.

I have done different tests. With either of the two settings ("AgentPong": 60 or "AgentPing": 60), it is necessary to restart the client agent on all remote computers to set them online. Besides, when the ubuntu server is restarted (mesh server is instaled on ubuntu 18.04.5) all the computers set offline again.

On one of remote machine that causes problems I have stopped the service and executed the instruction:
meshagent run > agent-output.txt
A new device has been created on a MeshCentral with the same name as the old device. The agent-output.txt report shows the following:
MeshCentral2 Agent

Connecting to: wss://xx.com:445/agent.ashx
Connected.
Server verified meshcore... Launching meshcore...

The device is connected correctly, but if the server is restarted for any reason, software update ..., these computer have to be offline again.

These new lines are added to the report when the server is restarted:

Mesh Server Connection Error
Connecting to: wss://xx.com:445/agent.ashx
Mesh Server Connection Error
AutoRetry Connect in 1263 milliseconds

The server takes about three minutes to restart. The agent should retry the reconnection process in 1.2 seconds but the agent goes offline and does not reconnect. Does this process run once or does it retry?

I have tried to update to version 0.6.71 but with the same result.

As a temporary solution I have created a script to restart the Mesh service twice a day.

Thanks for help me.

**Comment by krayon007** [comment]
I think I root caused this... When we used LoadLibraryEx, we used the flag `LOAD_LIBRARY_SEARCH_SYSTEM32`, which I noticed on the MSDN documentation says: 

> Windows 7, Windows Server 2008 R2, Windows Vista and Windows Server 2008:  This value requires KB2533623 to be installed.

Here is the [link](https://support.microsoft.com/en-us/help/2533623/microsoft-security-advisory-insecure-library-loading-could-allow-remot) to the KBB. I will update the agent to work around this, as the original reason for using this in our timer call (GetTickCount64) was because it wasn't supported on Windows XP, so we needed to detect if we needed to fallback, however, since we no longer support WinXP, this isn't needed anymore, as I can just call GetTickCount64() directly.

The bug, was that if this KBB wasn't installed on Windows 7 SP1 or older or Windows Server 2008 R2, this call would fail, which would result in the timer returning 0, which would cause a busy loop. This is why the CPU % would sometimes go to 100%. And since the GetTickCount64() call failed in this case, timeouts would not be set correctly, which is why the agent would disconnect after a while, becuase it isn't sending out keep alives correctly.

The easiest way to fix this issue without waiting for an update, is to manually install the KBB referenced above
